plugins/reach/lib/kReachManager.php

```diff
@@ -3,8 +3,130 @@
 /**
  * @package plugins.reach
  */
-class kReachManager implements kObjectChangedEventConsumer, kObjectCreatedEventConsumer, kObjectAddedEventConsumer
+class kReachManager implements kObjectChangedEventConsumer, kObjectCreatedEventConsumer, kObjectAddedEventConsumer, kGenericEventConsumer
 {
+	/**
+	 * @var array<booleanNotificationTemplate>
+	 */
+	protected $booleanNotificationTemplates;
+
+	protected $gotBooleanNotificationTemplates;
+	protected $gotEmptyBooleanNotificationTemplates;
+
+	protected function getObjectType($eventObjectClassName)
+	{
+		switch($eventObjectClassName)
+		{
+			case "entry":
+				return  objectType::ENTRY;
+			case "category":
+				return objectType::CATEGORY;
+			case "asset":
+				return objectType::ASSET;
+			case "flavorAsset":
+				return objectType::FLAVORASSET;
+			case "thumbAsset":
+				return objectType::THUMBASSET;
+			case "uiconf":
+				return  objectType::UICONF;
+			case "conversionProfile2":
+				return objectType::CONVERSIONPROFILE2;
+			case "kuser":
+				return objectType::KUSER;
+			case "permission":
+				return objectType::PERMISSION;
+			case "permissionItem":
+				return  objectType::PERMISSIONITEM;
+			case "userRole":
+				return  objectType::USERROLE;
+			default:
+				return null;
+		}
+	}
+	/* (non-PHPdoc)
+	 * @see kGenericEventConsumer::consumeEvent()
+	 */
+	public function consumeEvent(KalturaEvent $event)
+	{
+		$scope = $event->getScope();
+		$partnerId = $scope->getPartnerId();
+		$object = $scope->getObject();
+		$entryId = $object->getEntryId();
+		if (count($this->booleanNotificationTemplates) == 0)
+			return false;
+		foreach ($this->booleanNotificationTemplates as $booleanNotificationTemplate)
+		{
+			$profileId = $booleanNotificationTemplate[0];
+			$booleanNotificationTemaplateObjects = $booleanNotificationTemplate[1];
+			$fullFieldCatalogItemIds = $booleanNotificationTemaplateObjects->getCatalogItemIds();
+			$allowedCatalogItemIds = PartnerCatalogItemPeer::retrieveActiveCatalogItemIds($fullFieldCatalogItemIds, $partnerId);
+			if(!count($allowedCatalogItemIds))
+			{
+				KalturaLog::debug("None of the fullfield catalog item ids are active on partner, [" . implode(",", $fullFieldCatalogItemIds) . "]");
+				continue;
+			}
+			$existingCatalogItemIds = EntryVendorTaskPeer::retrieveExistingTasksCatalogItemIds($entryId, $allowedCatalogItemIds);
+			$catalogItemIdsToAdd = array_unique(array_diff($allowedCatalogItemIds, $existingCatalogItemIds));
+			foreach ($catalogItemIdsToAdd as $catalogItemIdToAdd)
+			{
+				//Pass the object Id as the context of the task
+				self::addEntryVendorTaskByObjectIds($entryId, $catalogItemIdToAdd, $profileId, $this->getContextByObjectType($object));
+			}
+		}
+		return true;
+	}
+
+	/* (non-PHPdoc)
+	 * @see kGenericEventConsumer::shouldConsumeEvent()
+	 */
+	public function shouldConsumeEvent(KalturaEvent $event)
+	{
+		$this->booleanNotificationTemplates = array();
```

Comment:
Run this code only once in init method that will generate static associative array that will contain the following info - key (object type , event type) -> value( profile id , catelogItem , boolean event notification ID).
This will running the loop multiple times.
Eventully , you just need to check if the object type and and event type are in the array